op-node: Initialize EngineController inside the Driver

trianglesphere opened the pull request:
**Description**

This moves the EngineController up to be able to use it without having to intialize the Derivation Pipeline.

**Tests**

No tests.

**Metadata**

- Fixes https://github.com/ethereum-optimism/client-pod/issues/448

ajsutton commented on op-node/rollup/driver/state.go:
```diff
@@ -556,8 +561,8 @@ type hashAndErrorChannel struct {
 func (s *Driver) checkForGapInUnsafeQueue(ctx context.Context) error {
-	start := s.derivation.UnsafeL2Head()
-	end := s.derivation.UnsafeL2SyncTarget()
+	start := s.engineController.UnsafeL2Head()
+	end := s.derivation.HighestUnsafeBlock()
```
I think this is what's breaking the P2P alt sync tests. Previously `UnsafeL2SyncTarget` was the lowest pending unsafe block, allowing this function to calculate the gap between the current L2 unsafe head and the next unsafe block we had to import.  Now that it's the highest block we are trying to rerequest all the pending blocks we already have rather than just filling the gap.

trianglesphere replied:
that seems like it'd be it. I'm dumb & was kinda randomly patching stuff up. This also likely breaks the assumption of what syncing was supposed to do.

trianglesphere replied:
Fixed this. It also simplified things which was nice.

ajsutton reviewed (commented):
Looks good except for the noted change that I think is breaking p2p alt sync.

trianglesphere committed:
op-node: Initialize EngineController inside the Driver

This moves the EngineController up to be able to use it without having
to intialize the Derivation Pipeline.

ajsutton reviewed (approved):
LGTM.
